lms/djangoapps/certificates/models.py

```diff
@@ -236,6 +237,25 @@ def invalidate(self):
         self.save()
 
 
+class CertificateGenerationHistory(TimeStampedModel):
+    """
+    Model for storing Certificate Generation History.
+    """
+
+    course_id = CourseKeyField(max_length=255)
+    generated_by = models.ForeignKey(User)
+    instructor_task = models.ForeignKey(InstructorTask)
+    is_regeneration = models.BooleanField(default=False)
+
+    class Meta(object):
+        app_label = "certificates"
+
+    def __unicode__(self):
+        return u"{'id': %(id)s, 'Course': %(course)s, 'Task': %(task)s}" % dict(
```

Comment:
I think its not a good idea to overload `__unicode__` with so much stuff. I think it should be something like this
`return u"certificates %s by %s on %s for %s" % ("regenerated" if self. is_regeneration else "generated", self.generated_by, self.create, self.course_id)`